packages/gatsby/src/redux/nodes.ts

```diff
@@ -133,27 +142,26 @@ const getResolvedNode = (typeName, id) => {
   return node
 }
 
-exports.getResolvedNode = getResolvedNode
-
-const addResolvedNodes = (typeName, arr) => {
+export const addResolvedNodes = (typeName: string): INode[] => {
+  const resolvedNodes: INode[] = []
   const { nodesByType, resolvedNodesCache } = store.getState()
-  const nodes /*: Map<mixed> */ = nodesByType.get(typeName)
+  const nodes = nodesByType.get(typeName)
 
   if (!nodes) {
-    return
+    return resolvedNodes
   }
 
-  const resolvedNodes = resolvedNodesCache.get(typeName)
+  const resolvedNodesFromCache = resolvedNodesCache.get(typeName)
 
   nodes.forEach(node => {
-    if (resolvedNodes) {
-      node.__gatsby_resolved = resolvedNodes.get(node.id)
+    if (resolvedNodesFromCache) {
+      node.__gatsby_resolved = resolvedNodesFromCache.get(node.id)
     }
-    arr.push(node)
+    resolvedNodes.push(node)
   })
-}
 
-exports.addResolvedNodes = addResolvedNodes
+  return resolvedNodes
+}
```

Comment:
This function was changed to create an array and return it instead of modifying the argument array. Having an idempotent function feels better than having one with side effects. This function was only used in one place, and its usage was identical to this: 

https://github.com/gatsbyjs/gatsby/pull/21010/files#diff-5d2176cb542d5c01d884139970be33fbR314

Reply:
Need to check this. Could be faster, actually. Could also be slower. (Consider the case where each type has 100k nodes and they get merged. Before they were pushed onto the same node array, after they are pushed in their own array and later merged through a `_.flatten()` (which should really just be a `.concat()` imo)).

Reply:
I checked this. It's slower, for sure, but in the case of the 150k ifsc benchmark it adds 5 to 10 seconds. I don't like slowing the build down for the user for the sake of something that only affects us at dev time, but I think the right choice is to do it anyways in this case ... :/